(b)ski: kernel boot broken after recent change (i.e. 3279e9d)

A user reported a problem with bski, where the kernel boot hang early. I could reproduce that now lcoally with a different kernel, too:

```
root@b450m-g-p:~/ski-test-nfs# screen ./bski-test.bash -noconsole /usr/src/ski/ski-bootloader/ski-bootloader /boot/vmlinuz-7.0.0-fa6053e85-ia64-ski root=/dev/sda simscsi=./sd simeth=tap0 init=/init PATH=/bin:/sbin:/usr/bin:/usr/sbin rw

-noconsole
/usr/src/ski/ski-bootloader/ski-bootloader
loading /tmp/bski.mn2TaJsNea/vmlinuz-7.0.0-fa6053e85-ia64-ski...
probing initramfs ...
initramfs not passed
starting kernel...
[    0.000000] Linux version 7.0.0-fa6053e85-ia64-ski (root@b450m-g-p) (ia64-linux-gcc (GCC) 16.0.1 20260412 (experimental), GNU ld (GNU Binutils) 2.46) #1 SMP PREEMPT Mon Apr 13 01:05:44 CEST 2026
[    0.000000] efi: EFI v1.0 by Hewlett-Packard
[    0.000000] efi: SALsystab=0x10a3f0 
[    0.000000] warning: unable to switch EFI into virtual mode (status=131)
[    0.000000] No I/O port range found in EFI memory map, falling back to AR.KR0 (0xffffc000000)
[    0.000000] printk: legacy console [simcons0] enabled
[    0.000000] SAL 0.1: Hewlett-Packard HP-simulator version 0.0
[    0.000000] get_cache_info: ia64_pal_cache_summary() failed (status=-1)
[    0.000000] PAL_VM_PAGE_SIZE failed with status=-1; defaulting to architected purge page-sizes.
[    0.000000] MCA related initialization done
[    0.000000] Zone ranges:
[    0.000000]   DMA32    [mem 0x0000000000000000-0x00000000ffffffff]
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x00000000c5ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x00000000c5ffffff]
[    0.000000] On node 0, zone DMA32: 14848 pages in unavailable ranges
[    0.000000] Kernel command line: root=/dev/sda simscsi=./sd simeth=tap0 init=/init PATH=/bin:/sbin:/usr/bin:/usr/sbin rw 
[    0.000000] Unknown kernel command line parameters "PATH=/bin:/sbin:/usr/bin:/usr/sbin", will be passed to user space.
[    0.000000] printk: log buffer data + meta data: 65536 + 229376 = 294912 bytes
[    0.000000] Dentry cache hash table entries: 524288 (order: 6, 4194304 bytes, linear)
[    0.000000] Inode-cache hash table entries: 262144 (order: 5, 2097152 bytes, linear)
[    0.000000] Sorting __ex_table...
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 50688
[    0.000000] mem auto-init: stack:all(zero), heap alloc:off, heap free:off
## here it hangs (forever). `bski` also keeps hogging a full hardware thread on the host.
## For a successful boot log up until where an init script takes over, see next comment.
```

> **NOTE:** `bski(-test).bash` is just a wrapper that allows me to use compressed kernel images with bski, by unpacking them to a temporary location and start them from there. `[...]-test[...]` because it has the path to `bski` hardcoded to the `bski` created during each bisect step.

...and bisected this to:

```
root@b450m-g-p:/usr/src/ski# git bisect good
3279e9defb3b2a2c7cc76818abc4c80aadbb9343 is the first bad commit
commit 3279e9defb3b2a2c7cc76818abc4c80aadbb9343
Author: Sergei Trofimovich <slyich@gmail.com>
Date:   Sun Apr 5 22:42:34 2026 +0100

    src/tlb.c: fix `Accesstype` vs `Accesstypemask` confsion
    
    Noticed by `clang`:
    
    ```
    ski> tlb.c:698:16: warning: comparison of different enumeration types ('Accesstype' and 'Accesstypemask') [-Wenum-compare]
    ski>   698 |             if (atype != READ_ACCESS || cpl > pl)
    ski>       |                 ~~~~~ ^  ~~~~~~~~~~~
    ski> tlb.c:734:16: warning: comparison of different enumeration types ('Accesstype' and 'Accesstypemask') [-Wenum-compare]
    ski>   734 |             if (atype == READ_ACCESS && cpl)
    ski>       |                 ~~~~~ ^  ~~~~~~~~~~~
    ```
    
    THe code should nto change, but it should be less confusing.

 src/tlb.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

root@b450m-g-p:/usr/src/ski# git bisect log
git bisect start
# status: waiting for both good and bad commits
# good: [ad834c087ac96cf464c73f06051c9ae089fe0d1c] Merge pull request #28 from linux-ia64/memory-map-and-processor-id-updates
git bisect good ad834c087ac96cf464c73f06051c9ae089fe0d1c
# status: waiting for bad commit, 1 good commit known
# bad: [6984f89d0d529c8b752f03d3d623f57033d00e58] src/tlb.c: added more braces around nested `if`s
git bisect bad 6984f89d0d529c8b752f03d3d623f57033d00e58
# good: [da7b78c905d499fc3aa2bec75769ca5fb65d5ae7] src/tlb.c: disambiguate dangling nested if/else
git bisect good da7b78c905d499fc3aa2bec75769ca5fb65d5ae7
# bad: [3279e9defb3b2a2c7cc76818abc4c80aadbb9343] src/tlb.c: fix `Accesstype` vs `Accesstypemask` confsion
git bisect bad 3279e9defb3b2a2c7cc76818abc4c80aadbb9343
# good: [3507fdd41f4c23a4cf60135d4421e531391b3c63] src/load.c: don't return bools instead of pointers
git bisect good 3507fdd41f4c23a4cf60135d4421e531391b3c63
# first bad commit: [3279e9defb3b2a2c7cc76818abc4c80aadbb9343] src/tlb.c: fix `Accesstype` vs `Accesstypemask` confsion
```

For me that commit works, when the assignment is done as before and an extra step is used to set `atypem`.
```diff
diff --git a/src/tlb.c b/src/tlb.c
index 3c01097..3124402 100644
--- a/src/tlb.c
+++ b/src/tlb.c
@@ -691,7 +691,8 @@ static int keyCheck(Accesstype atype, unsigned key)
 static BOOL accessRights(unsigned ar, unsigned pl, unsigned cpl,
                         Accesstype atype)
 {
-    Accesstypemask atypem = atype & READ_ACCESS | WRITE_ACCESS;
+    atype &= READ_ACCESS | WRITE_ACCESS;
+    Accesstypemask atypem = atype;
 
     switch (ar) {
        case 0:
```
...but I guess the problem is actually that `&` takes precedence over `|`, so `atype & READ_ACCESS | WRITE_ACCESS` results in something different than `atype & (READ_ACCESS | WRITE_ACCESS)` which would be the same as the code before 3279e9d. So it should actually be doable in one line, too. I'll provide a PR with the fix.

--- a/src/tlb.c
+++ b/src/tlb.c
@@ -691,7 +691,7 @@ static int keyCheck(Accesstype atype, unsigned key)
 static BOOL accessRights(unsigned ar, unsigned pl, unsigned cpl,
 			 Accesstype atype)
 {
-    Accesstypemask atypem = atype & READ_ACCESS | WRITE_ACCESS;
+    Accesstypemask atypem = atype & (READ_ACCESS | WRITE_ACCESS);
 
     switch (ar) {
 	case 0:

